api/api.go

```diff
@@ -613,6 +613,18 @@ func (b *BlockChainAPI) prepareBlockResponse(
 		Timestamp:    hexutil.Uint64(block.Timestamp),
 	}
 
+	// todo remove after previewnet, temp fix to mock some of the timestamps
+	if block.Timestamp == 0 {
+		first := uint64(1715189257)
+		blockTime := uint64(200)
+		firstRecordedTimestampBlock := uint64(5493)
+
+		diff := firstRecordedTimestampBlock - block.Height
+		timestamp := first - blockTime*diff
```

Comment:
It should be - because what I did is checked what is the first timestamp we have indexed, which is: `1715189257` then I subtract `200` (average block timestamp change I calculated) as many times as there is a diff from the height of the block that has this value and the block we are generating. Basically height 0 would have biggest change and height `5492` which is one less than first recorded timestamp would have lowest change

Reply:
I could have better documented, but this will all go away